File: code/back-office/front/src/app/dto/election.ts
import { ElectionType } from './election-type';

export class Election {
  id: number;
  type: ElectionType;
  name: string;
  startDate: string;
  endDate: string;
  pollingStationCount: number;
  totalPollingStationCount: number;
  status: number;

  constructor(
    id: number,
    type: ElectionType,
    name: string,
    startDate: string,
    endDate: string,
    pollingStationCount: number,
    totalPollingStationCount: number,
    status: number
  ) {
    this.id = id;
    this.type = type;
    this.name = name;
    this.startDate = startDate;
    this.endDate = endDate;
    this.pollingStationCount = pollingStationCount;
    this.totalPollingStationCount = totalPollingStationCount;
    this.status = status;
  }
}

File: code/back-office/server/src/main/java/ceni/system/votesync/model/entity/election/Election.java
package ceni.system.votesync.model.entity.election;

import java.sql.Date;

import ceni.system.votesync.model.base.entity.AbstractEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
@Entity
@Table(name = "elections")
public class Election extends AbstractEntity<Integer> {

    @OneToOne
    @JoinColumn(name = "id_type_election")
    private ElectionType type;

    @Column(name = "nom")
    private String name;

    @Column(name = "date_debut")
    private Date startDate;

    @Column(name = "date_fin")
    private Date endDate;

    @Column(name = "etat")
    private Integer status;
}
